redaxo/src/addons/structure/plugins/content/lib/api_module.php

```diff
@@ -8,53 +8,80 @@ class rex_module
     /**
      * @var int
      */
-    private $module_id;
+    private $id;
     /**
      * @var string|null
      */
     private $key;
 
     public function __construct(int $module_id)
     {
-        $this->module_id = $module_id;
+        $this->id = $module_id;
         $this->key = '';
     }
 
     public static function forKey(string $module_key): ?self
     {
-        $sql = rex_sql::factory();
-        $sql->setQuery('select id from '. rex::getTable('module') .' where `key`=?', [$module_key]);
+        $mapping = self::getKeyMapping();
 
-        if (1 == $sql->getRows()) {
-            $module_id = $sql->getValue('id');
+        if (false !== $id = array_search($module_key, $mapping, true)) {
+            $module = new self($id);
+            $module->key == $module_key;
 
-            $module = new self($module_id);
-            $module->key = $module_key;
             return $module;
         }
+
         return null;
     }
 
     public function getId(): int
     {
-        return $this->module_id;
+        return $this->id;
     }
 
     public function getKey(): ?string
     {
         // key will never be empty string in the db
         if ('' === $this->key) {
-            $sql = rex_sql::factory();
-            $sql->setQuery('select `key` from '. rex::getTable('module') .' where id=?', [$this->module_id]);
-
-            if (1 == $sql->getRows()) {
-                $this->key = $sql->getValue('key');
-            } else {
-                $this->key = null;
-            }
+            $this->key = self::getKeyMapping()[$this->id] ?? null;
             assert('' !== $this->key);
         }
 
         return $this->key;
     }
+
+    public static function deleteKeyMappingCache(): void
+    {
+        rex_file::delete(self::getKeyMappingPath());
```

Comment:
Cache handling haben wie sonst in service klassen nicht direkt in api klasse

Reply:
Allerdings hat `rex_template` sowieso schon eine deleteCache-Methode, und die Klassen sind schon etwas anders vom Aufbau (zum Beispiel auch kein `::get($id)`, sondern aufruf über den constructor für IDs).

Ich kann aber auch jeweils neue Cache-Klassen einführen.

Reply:
Ggf. Das rex_template cache verhalten deprecated und alles in eine cache klasse analog der anderen realisieren?